pkg/aws/helpers.go

```diff
@@ -1000,3 +991,63 @@ func IsHostedCPManagedPolicies(cluster *cmv1.Cluster) bool {
 func IsHostedCP(cluster *cmv1.Cluster) bool {
 	return cluster.Hypershift().Enabled()
 }
+
+func ConvertToTagPointers(tags []iamtypes.Tag) []*iamtypes.Tag {
```

Comment:
Do we need this conversation? 
Previously `ListPolicyTagsInput` returned a slice of pointers to tags, with V2 it returns a slice of tags.
I wonder if we should change the signature of `isAccountRoleVersionCompatible` and `IamResourceHasTag` to get a slice of tags, and then we can save the conversion and reduce the number of pointers in the code.

Reply:
should the same be done for: `HasCompatibleVersionTags` ?

Reply:
Since `HasCompatibleVersionTags` is already in the ocm-common, to avoid the delay of this PR, I assume it's ok to refactor it in a follow-up PR.

Reply:
@den-rgb we can now remove `ConvertToTagPointers` as it's an unused function.